File: Kotlin/1047.kt
import java.util.*

fun main(args: Array<String>) {
    val scanner = Scanner(System.`in`)

    val h1 = scanner.nextInt()
    val m1 = scanner.nextInt()
    var h2 = scanner.nextInt()
    var m2 = scanner.nextInt()

    var t: Int
    var h: Int
    var m: Int

    if (h2 <= h1 && m2 <= m1) {
        h2 += 24
    } else if (m2 <= m1) {
        m2 += 60
        h2 -= 1
    }

    t = ((h2 * 3600) + (m2 * 60)) - ((h1 * 3600) + (m1 * 60))
    h = t / 3600
    m = (t - (h * 3600)) / 60

    println("O JOGO DUROU $h HORA(S) E $m MINUTO(S)")
}

File: C++/1047.cpp
#include <iostream>
using namespace std;

int main(){
    int h1, m1, h2, m2, t, h, m;

    cin >> h1 >> m1 >> h2 >> m2;

    if (h2 <= h1 && m2 <= m1){
        h2 = h2 + 24;
    
    }else if (m2 <= m1){
        m2 = m2 + 60;
        h2 = h2 - 1;
    }

    t = ((h2 * 3600) + (m2 * 60)) - ((h1 * 3600) + (m1 * 60));
    h = t / 3600;
    m = (t - (h * 3600)) / 60;

    cout << "O JOGO DUROU " << h << " HORA(S) E " << m << " MINUTO(S)" << endl;
    return 0;
}